test/e2e/storage/csi_mock/csi_volume_expansion.go

```diff
@@ -201,6 +201,130 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 			})
 		}
 	})
+	ginkgo.Context("CSI online volume expansion with secret[Feature:CSINodeExpandSecret]", func() {
+		var stringSecret = map[string]string{
+			"username": "admin",
+			"password": "t0p-Secret",
+		}
+		trackedCalls := []string{
+			"NodeExpandVolume",
+		}
+		tests := []struct {
+			name          string
+			disableAttach bool
+			expectedCalls []csiCall
+
+			// Called for each NodeExpandVolume calls, with counter incremented atomically before
+			// the invocation (i.e first value will be 1).
+			nodeExpandHook func(counter int64) error
+		}{
+			{
+				name: "should expand volume without restarting pod if attach=on, nodeExpansion=on, csiNodeExpandSecret=on",
+				expectedCalls: []csiCall{
+					{expectedMethod: "NodeExpandVolume", expectedError: codes.OK, expectedSecret: stringSecret},
```

Comment:
Can you add some negative test cases?

Reply:
of course

Reply:
Pass the secret and to check the secret was used by csi mock driver?